core/domain/classifier_services.py

```diff
@@ -370,51 +468,53 @@ def get_classifier_from_exploration_attributes(exp_id, exp_version,
             belongs.
 
     Returns:
-        ClassifierData. Domain object for the Classifier model.
+        ClassifierTrainingJob. Domain object for the Classifier training job
+            model.
 
     Raises:
-        Exception: Entity for class ClassifierExplorationMapping with id not
+        Exception: Entity for class TrainingJobExplorationMapping with id not
             found.
-        Exception: Entity for class ClassifierData with id not found.
+        Exception: Entity for class ClassifierTrainingJob with id not found.
     """
-    classifier_exploration_mapping_model = (
-        classifier_models.ClassifierExplorationMappingModel.get_model(
+    training_job_exploration_mapping_model = (
+        classifier_models.TrainingJobExplorationMappingModel.get_model(
             exp_id, exp_version, state_name))
-    classifier_id = classifier_exploration_mapping_model.classifier_id
-    classifier = get_classifier_by_id(classifier_id)
-    return classifier
+    if training_job_exploration_mapping_model is None:
+        return None
+    job_id = training_job_exploration_mapping_model.job_id
+    classifier_training_job = get_classifier_training_job_by_id(job_id)
+    return classifier_training_job
```

Comment:
Just a thought: as mentioned before, it would be nice if the classifier data that got computed could somehow be stored on the training job, so that we don't need a third GET to retrieve it.